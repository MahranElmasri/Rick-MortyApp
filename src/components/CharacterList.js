import Character from "./Character";
import React from "react";
import "./CharacterList.css";
import PropTypes from "prop-types";

export default function CharacterList({ character, loading }) {
  return (
    <div className="grid-4">
      {character.map((char) => {
        return <Character key={char.id} char={char} loading={loading} />;
      })}
    </div>
  );
}

CharacterList.propTypes = {
  charactar: PropTypes.object,
  loading: PropTypes.bool.isRequired,
};
