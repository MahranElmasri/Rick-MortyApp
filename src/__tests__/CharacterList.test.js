import CharacterList from "../components/CharacterList";

const characters = [
  {
    id: 1,
    image: "",
    name: "Rick Sanchez",
    gender: "Male",
    species: "Human",
    status: "Alive",
    created: "2017-11-04",
  },
  {
    id: 2,
    image: "",
    name: "Morty Smith",
    gender: "Male",
    species: "Human",
    status: "Alive",
    created: "2017-11-04",
  },
];

it("render corectlly", () => {
  const wrapper = shallow(
    <CharacterList character={characters} loading={false} />
  );

  expect(wrapper).toMatchSnapshot();
});

jest.mock("../service/fetchCharacters.js");

it("should fetch characters", (done) => {
  const wrapper = shallow(
    <CharacterList character={characters} loading={false} />
  );
  setTimeout(() => {
    wrapper.update();
    expect(wrapper.find("Character").length).toEqual(2);
    done();
  });
});
