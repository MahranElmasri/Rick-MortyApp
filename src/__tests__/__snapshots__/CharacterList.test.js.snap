// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render corectlly 1`] = `
<div
  className="grid-4"
>
  <Character
    char={
      Object {
        "created": "2017-11-04",
        "gender": "Male",
        "id": 1,
        "image": "",
        "name": "Rick Sanchez",
        "species": "Human",
        "status": "Alive",
      }
    }
    key="1"
    loading={false}
  />
  <Character
    char={
      Object {
        "created": "2017-11-04",
        "gender": "Male",
        "id": 2,
        "image": "",
        "name": "Morty Smith",
        "species": "Human",
        "status": "Alive",
      }
    }
    key="2"
    loading={false}
  />
</div>
`;
